charts/backstage/values.yaml

```diff
@@ -149,7 +149,7 @@ upstream:
         secret:
           defaultMode: 420
           optional: true
-          secretName: dynamic-plugins-npmrc
+          secretName: '{{ printf "%s-dynamic-plugins-npmrc" .Release.Name }}'
```

Comment:
I think this might break users relying on the current product docs to specify a custom NPM registry for dynamic plugin packages: https://docs.redhat.com/en/documentation/red_hat_developer_hub/1.2/html-single/configuring_plugins_in_red_hat_developer_hub/index#proc-using-custom-npm-registry

Since this optional secret is not managed by the Helm Chart (only used if it exists in the current namespace), I think it should be safe to be used by multiple Helm releases.

Reply:
Can't we update the docs?

Reply:
Sure, we should also update the docs.
I was thinking about existing customers already mounting their custom `dynamic-plugins-npmrc` Secret. I guess this change should also be documented as a potential breaking change in 1.3. And the Chart version should ideally also be bumped to 2.x, I think.
Which is why I was wondering if it was worth it, since this Secret is not managed by the Helm Chart, so resource naming should not be an issue here (unlike resources managed by the Helm Chart).

Reply:
@rm3l The user has updated the JIRA for this pull request with additional information about why they are requesting it. And I do agree with them that everything created/managed by the helm chart should be self-contained and named appropriately. I think that we should move ahead with this update and we can document the change. Also, since it is an optional secret it may very well not be used by many at this point since it is not automatically created, it is just mounted if it exists.

Reply:
Thanks for the context. I wasn't aware of that [`extraDeploy` field](https://github.com/backstage/charts/blob/main/charts/backstage/values.yaml#L38-L39) in the upstream Backstage Chart, but indeed, if they are using that to have this Secret created, then it becomes managed by the Helm Chart.
LGTM then, and let's update the docs.